composer/trainer/dist_strategy.py

```diff
@@ -369,78 +369,141 @@ def sync_hook(*args):
             if hasattr(obj, '_fsdp_wrap') and not bool(obj._fsdp_wrap):
                 continue
 
-            def _param_init_fn(module: torch.nn.Module) -> None:
-                # A dictionary of all tied parameter pointers to module names
+            # Rather than verifying these changes with older PyTorch versions, we are fixing forward here
+            if version.parse(torch.__version__) > version.parse('2.1.0'):
+                # A dictionary of all tied parameter pointers to (module, attr) tuples
                 tied_pointers = {}
 
                 # Goes through all modules finding which weights have the same pointers
-                for name, mod in module.named_modules():
-                    # Since FSDP recursively wraps, at parent modules we can encounter already
-                    # wrapped weights, as a result we should skip any modules with `_fsdp_wrapped_module.`
-                    if '_fsdp_wrapped_module' in name:
-                        continue
+                for _, mod in obj.named_modules():
                     for attr in ['weight', 'bias']:
                         if hasattr(mod, attr):
                             mod_attr = getattr(mod, attr)
                             if mod_attr is None:
                                 continue
```

Comment:
```suggestion
                for mod in obj.modules():
                    for attr, mod_attr in mod.named_parameters(recurse=False):
```

@bcui19 (?; or whoever originally wrote this) can you verify that this is fine?

Reply:
I'm guessing there is a reason we only did weight/bias? @bcui19

Reply:
Unfortunately, I don't remember if there was a particular reason we limited it to 'weight', 'bias'.

Reply:
Given that, I'd prefer to not to change it, since we know the current version works. okay with the @vchiley ?

Reply:
weird but 🤷‍♂️ 

Is it possible to include a warning or something that takes param names that are not 'weight' or 'bias' and say: "Warning: these cannot be tied".

Reply:
maybe i should just change it...

Reply:
I'm going to change it to your version for the new code, will leave the old code untouched.